rbi/core/module.rbi

```diff
@@ -132,7 +132,13 @@ class Module < Object
     )
     .returns(T.untyped)
   end
-  def class_eval(arg0, filename=T.unsafe(nil), lineno=T.unsafe(nil)); end
+  sig do
+    type_parameters(:U).params(
+        blk: T.proc.params(m: Module).returns(T.type_parameter(:U)),
```

Comment:
I have minor concern that `Module` type here it too restrictive(it doesn't use type of specific sublcass and thus singleton methods aren't available in scope) and will prohibit most code from being typed without introducing much value.

WDYT about using `T.untyped` to ease adoption?